tools/PI/DevHome.PI/ViewModels/BarWindowViewModel.cs

```diff
@@ -90,11 +98,22 @@ public BarWindowViewModel()
         }
 
         CurrentSnapButtonText = IsSnapped ? _UnsnapButtonText : _SnapButtonText;
+
+        _snapHelper = new();
     }
 
     partial void OnIsSnappedChanged(bool value)
     {
-        CurrentSnapButtonText = IsSnapped ? _UnsnapButtonText : _SnapButtonText;
+        if (IsSnapped)
+        {
+            CurrentSnapButtonText = _UnsnapButtonText;
+            _snapHelper.Snap();
```

Comment:
>_snapHelper.Snap(); [](http://example.com/codeflow?start=12&length=19)

Just a nit, but I'd suggest performing these snap actions in the PerformSnap() function, vs here (after we've advertised that snapping has completed).